src/components/navigation/navigation.js

```diff
@@ -25,21 +26,63 @@ export default class NavigationToggle {
     openNav() {
         const input = [...this.navigation.getElementsByTagName('INPUT')][0];
 
-        this.toggle.setAttribute(attrExpanded, 'true');
         this.toggle.classList.add('active');
         this.navigation.setAttribute(attrHidden, 'false');
         this.navigation.classList.remove(this.hideClass);
 
         if (input) {
             input.focus();
         }
+
+        const searchToggleBtn = document.querySelector('.ons-js-toggle-header-search');
+        if (this.toggle === searchToggleBtn) {
+            this.updateSearchIcon(true, this.toggle);
+        }
+
+        this.toggle.setAttribute(attrExpanded, 'true');
+
+        this.toggleMenuAndSearch();
     }
 
     closeNav() {
-        this.toggle.setAttribute(attrExpanded, 'false');
         this.toggle.classList.remove('active');
         this.navigation.setAttribute(attrHidden, 'true');
         this.navigation.classList.add(this.hideClass);
+
+        const searchToggleBtn = document.querySelector('.ons-js-toggle-header-search');
+        if (this.toggle === searchToggleBtn) {
+            this.updateSearchIcon(false, this.toggle);
+        }
+
+        this.toggle.setAttribute(attrExpanded, 'false');
+    }
+
+    updateSearchIcon(isOpen, toggle) {
+        const searchIconSVG = `
+            <span class="ons-btn__inner">
+                <svg class="ons-icon ons-icon--search ons-u-mr-2xs" viewBox="0 0 12 12" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" role="img" title="ons-icon-search">
+                    <path d="M11.86 10.23 8.62 6.99a4.63 4.63 0 1 0-6.34 1.64 4.55 4.55 0 0 0 2.36.64 4.65 4.65 0 0 0 2.33-.65l3.24 3.23a.46.46 0 0 0 .65 0l1-1a.48.48 0 0 0 0-.62Zm-5-3.32a3.28 3.28 0 0 1-2.31.93 3.22 3.22 0 1 1 2.35-.93Z"></path>
+                </svg>
+                <span class="ons-btn__text"></span>
+            </span>`;
+
+        const closeIconSVG = `
+            <span class="ons-btn__inner">
+                <svg class="ons-icon ons-icon--close" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" focusable="false" fill="currentColor" role="img" title="ons-icon-close">
+                    <path d="M12 0C5.4 0 0 5.4 0 12C0 18.6 5.4 24 12 24C18.6 24 24 18.6 24 12C24 5.4 18.6 0 12 0ZM17.3143 15.5143C17.6571 15.8571 17.6571 16.3714 17.3143 16.7143L16.7143 17.3143C16.3714 17.6571 15.8571 17.6571 15.5143 17.3143L12 13.8L8.48571 17.3143C8.14286 17.6571 7.62857 17.6571 7.28571 17.3143L6.68571 16.7143C6.34286 16.3714 6.34286 15.8571 6.68571 15.5143L10.2 12L6.68571 8.48571C6.34286 8.14286 6.34286 7.62857 6.68571 7.28571L7.28571 6.68571C7.62857 6.34286 8.14286 6.34286 8.48571 6.68571L12 10.2L15.5143 6.68571C15.8571 6.34286 16.3714 6.34286 16.7143 6.68571L17.3143 7.28571C17.6571 7.62857 17.6571 8.14286 17.3143 8.48571L13.8 12L17.3143 15.5143Z"/>
+                </svg>
+                <span class="ons-btn__text"></span>
+            </span>`;
+
+        if (isOpen) {
+            toggle.classList.add('ons-btn--close');
```

Comment:
This is added first time you click the button and then kept, do we need this anymore? The css attached doesn't seem to do anything